### PR TITLE
Fix EventSource to stop ignoring EventCommand.SendManifest

### PR DESCRIPTION
**Pull request by sywhang**
EventSource was ignoring SendManifest commands coming in from ETW specifically, because the part of the code that was sending manifest data over was ifdef'd out. Normally we send the manifest data on enabling a session, so this path was never hit. But if another manifest data needs to be sent after the trace has already started (i.e. change the output file to a different file), it ignores that request.

### CODE REVIEW

File: src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs [hunk 1]
```diff
@@ -2773,14 +2773,12 @@ internal void DoCommand(EventCommandEventArgs commandArgs)
...
                 else
                 {
-#if !FEATURE_PERFTRACING
                     if (commandArgs.Command == EventCommand.SendManifest)
                     {
                         // TODO: should we generate the manifest here if we hadn't already?
                         if (m_rawManifest != null)
                             SendManifest(m_rawManifest);
                     }
-#endif
```
Comment (stephentoub) [inline src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs]: Is there a test we can write for this?

Comment (stephentoub) [reply src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs]: Also, did you audit the other uses of FEATURE_PERFTRACING to make sure the same issue doesn't exist elsewhere?

**Review by stephentoub — APPROVED** [review]

File: src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs (hunk 1 above)
Comment (sywhang) [reply src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs]: Thanks Stephen - I was initially going to add the test to dotnet/diagnostics repo but figured that it's probably better to add some basic validation tests so I added a couple of tests to this PR. 

And yes, I went through every other usage of FEATURE_PERFTRACING and confirmed that we don't have a similar issue.

File: src/libraries/System.Diagnostics.Tracing/tests/BasicEventSourceTest/TestsManifestGeneration.Etw.cs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+            {
+                using (RemoteInvokeHandle handle = RemoteExecutor.Invoke(() =>
+                {
+                    var es = new SimpleEventSource();
+                    for (var i = 0; i < 100; i++)
+                    {
+                        es.WriteSimpleInt(i);
+                        Thread.Sleep(100);
+                    }
+                }))
```
Comment (josalem) [inline src/libraries/System.Diagnostics.Tracing/tests/BasicEventSourceTest/TestsManifestGeneration.Etw.cs]: I'm not sure this is doing what you intend it to.  The code inside the lambda here won't run until `dispose` is called I think, which won't happen until after you hit the end of the `using` block and all your collection code has executed.  That would explain why the test is failing.

**Pull request merged**